elfo-network/src/config.rs

```diff
@@ -58,22 +58,46 @@ pub struct Config {
     pub idle_timeout: Duration,
 }
 
+/// Preference.
+#[derive(Debug, Clone, Copy, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Preference {
+    /// This is preferred.
+    Preferred,
+
+    /// This is just supported.
+    Supported,
+}
+
+/// Preferences for the compression algorithms
+/// selection.
+#[derive(Debug, Deserialize, Default, Clone)]
+pub struct CompressionPreference {
+    /// LZ4 compression algorithm.
+    pub lz4: Option<Preference>,
+}
+
 /// Compression settings.
 #[derive(Debug, Default, Deserialize, Clone)]
 pub struct CompressionConfig {
-    /// Compression algorithm.
+    /// Compression algorithms.
+    ///
+    /// Example:
+    /// ```toml
+    /// algorithms = { lz4 = "preferred" }
+    /// ```
+    ///
+    /// Preferred implies supported.
     #[serde(default)]
-    pub algorithm: CompressionAlgorithm,
+    pub algorithms: CompressionPreference,
```

Comment:
This is a breaking change (which is ok because we have alpha versions), but, in this case, I think we can revise it even more and use `system.network.compression.lz4 = "Preferred"`. If other settings (e.g. compression level) will be available to configure in the future, it will support the second per-codec form `system.network.compression.zstd = { preference = "Preferred", level = 6 }` in a backward-compatible way. I don't see what common options can be added next to `algorithms` to have this additional nesting level.

Reply:
Still "algorithms"

Reply:
gonna fix on my own